ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/ProjectImportOutputJsonRpcSubscriber.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.projectimport.wizard;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerConfigurator;
+import org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerManager;
+import org.eclipse.che.api.core.jsonrpc.commons.RequestTransmitter;
+import org.eclipse.che.api.project.shared.ImportProgressRecord;
+import org.eclipse.che.api.project.shared.dto.ImportProgressRecordDto;
+
+import java.util.function.Consumer;
+
+/**
+ * Json RPC subscriber for listening to the project import events. Register itself for the listening events from the server side.
+ *
+ * @author Vlad Zhukovskyi
+ * @since 5.9.0
+ */
+@Singleton
+public class ProjectImportOutputJsonRpcSubscriber {
+
+    public static final String WS_AGENT_ENDPOINT = "ws-agent";
+
+    private static final String EVENT_IMPORT_OUTPUT_SUBSCRIBE    = "event:import-project:subscribe";
```

Comment:
This probably should be changed to something like `importProject/subscribe` according to our json rpc naming conventions